BUGFiX: Handle zero as tag content properly

mficzel opened the pull request:
Tags that contained a zero as content (which can occur in for instance in select-options)
were parsed as tag with empty child list because of a weak comparison in the parser code that
treated a zero as falsy.

This change adjusts the comparison and adds tests for zero and space as attributes and contents.

mficzel committed:
BUGFiX: Handle zero as tag content properly

Tags that contained a zero as content (which can occur in for instance in select-options)
were parsed as tag with empty child list because of a weak comparison in the parser code that
treated a zero as falsy.

This change adjusts the comparison and adds tests for zero and space as attributes and contents.

gerhard-boden commented on Classes/Parser/Expression/NodeList.php:
```diff
@@ -34,7 +34,7 @@ public static function parse(Lexer $lexer): array
             if ($lexer->isOpeningBracket()) {
                 $lexer->consume();
-                if ($currentText) {
+                if ($currentText !== '') {
```
should we think about `trim()` here?

gerhard-boden replied:
What about https://github.com/neos/fusion-afx/pull/31/files#diff-5c2ca3faa504ec000342c9d5490af938R85?

mficzel replied:
@gerhard-boden what would trim() achieve here? Empty strings in the payload are handled in the afx-service and not in the payload. The service then decides on which whitespace is meaningful or not.

I agree that the second comparison below should be adjusted aswell.

gerhard-boden replied:
thanks for clarifying! Just saw the condition for the empty string and figured it could be possible that the payload contains whitespace, but as you pointed it's not possible and therefore nothing to worry about. Please ping me whenever the second condition is handled for my approval vote :+1:

mficzel committed:
BUGFIX: Remove second weak string check

This one did not cause bugs "yet" but has the same bad pattern as the
comparison on top that did not allow 0 as content.

mficzel commented:
@gerhard-boden adjusted the second comparison aswell. 

Before merging we should think about the target branch for a second:
- This change is for master and could be released as afx 1.4.1 
- It could be backported to older afx versions but afx 1.4.1 does support Neos 4.2 and upwards already so all supported neos versions are covered.
Thus i tend to only release 1.4.1 and do not mess with backporting this now.

gerhard-boden reviewed (approved):
thanks for the fix! 👍 
About the merge:
Sounds good to me, no need to backport IMHO.

Those affected by this bug can easily update this package and I guess nobody relies on this broken behavior 😂
